utils/dataconvert/dataconvert.cpp

```diff
@@ -3056,10 +3064,10 @@ int64_t DataConvert::intToTime(int64_t data, bool fromString)
 #if defined(__GNUC__) && __GNUC__ >= 7
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wformat-truncation="
-    snprintf( buf, 15, "%llu", (long long unsigned int)data);
+    snprintf( buf, 15, "%lld", (long long int)data);
```

Comment:
A bug I found while going through negative test cases. Since time can be negative this was causing a loss of sign.